ouroboros-consensus/src/ouroboros-consensus/Ouroboros/Consensus/MiniProtocol/ChainSync/Client.hs

```diff
@@ -580,6 +613,8 @@ data DynamicEnv m blk = DynamicEnv {
     -- ^ Ensure that the LoP bucket is leaking. Can be called on an
     -- already-leaking bucket.
   , grantLoPToken       :: m ()
+  , setTheirTip         :: Their (Tip blk) -> STM m ()
+  , setLatestSlot       :: SlotNo -> STM m ()
```

Comment:
Should the latest slot type be `WithOrigin SlotNo` instead? Otherwise, what is the value of the latest slot when no headers have been received?